### PR TITLE
WT-2307 debugging branch

### PR DESCRIPTION
**Pull request by keithbostic**
Debugging branch, never intended to be merged.

### CODE REVIEW

File: src/btree/bt_curnext.c [hunk 1]
```diff
@@ -431,6 +431,147 @@ __wt_btcur_iterate_setup(WT_CURSOR_BTREE *cbt)
...
+	    __wt_last_op_name(cbt->last_op[13]),
+	    __wt_last_op_name(cbt->last_op[14]),
+	    __wt_last_op_name(cbt->last_op[15]),
+	    __wt_last_op_name(cbt->last_op[16]),
+	    __wt_last_op_name(cbt->last_op[17]),
+	    __wt_last_op_name(cbt->last_op[18]),
+	    __wt_last_op_name(cbt->last_op[19]));
+
+	for (i = cbt->next_track - 1; i >= 0; --i)
+		__wt_errx(session,
```
Comment (RedBeard0531) [inline src/btree/bt_curnext.c]: Assuming that cursors are calloc'ed (or otherwize zero initialized) does it make sense to skip printing in these loops if .locate == 0? Or make __wt_track_location_name have a special case for 0?

Comment (keithbostic) [inline src/btree/bt_curnext.c]: They get set to 0 if the cursor doesn't ref a page -- I don't anticipate that happening, but it's a sanity check if it does.

**Comment by keithbostic** [comment]
@agorrod, @sueloverso, @daveh86: This branch was a debugging branch, and is no longer interesting. However, I would like to understand why key-out-of-order tests are flagging some of the standard tests (that's why it's not passing checks):
```
ERROR: test_cursor05.test_cursor05.test_without_colgroups
ERROR: test_cursor05.test_cursor05.test_with_colgroups
ERROR: test_txn01.test_read_committed_default.test_read_committed_default
```
Anyway, if somebody gets to it before I do, that'd be great, otherwise I'll get back to it.

Alex's branch #2407 fails in similar ways (it has the same checking code).

**Comment by sueloverso** [comment]
@keithbostic I can reproduce the error on these debug branches, but it is fixed now with the merge of #2415.

**Comment by keithbostic** [comment]
Deleting this diagnostic pull request for now, #2415 will hopefully resolve the problem.

**Pull request closed without merging**